.github/workflows/build.yml

```
@@ -39,33 +39,48 @@ jobs:
         uses: lukka/get-cmake@latest
         
       - name: Configure
         shell: bash
         run: |
-          if [ "${{ env.COMPILER }}" == "msvc" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc
-          elif [ "${{ env.COMPILER }}" == "clang" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CC=clang -DCMAKE_CXX=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu
-          elif [ "${{ env.COMPILER }}" == "gcc" ]; then
-            cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu
+          # TODO: for some reason windows has a different path for the generators folder than linux
+          if [ "${{ env.OPERATING_SYSTEM }}" == "linux" ]; then
+            if [ "${{ env.COMPILER }}" == "msvc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "clang" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CC=clang -DCMAKE_CXX=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "gcc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-gnu -DCMAKE_TOOLCHAIN_FILE=build/$BUILD_TYPE/generators/conan_toolchain.cmake
+            fi
+          elif [ "${{ env.OPERATING_SYSTEM }}" == "windows" ]; then
+            if [ "${{ env.COMPILER }}" == "msvc" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_C_COMPILER_TARGET=x86_64-pc-$OPERATING_SYSTEM-msvc -DCMAKE_TOOLCHAIN_FILE=build/generators/conan_toolchain.cmake
+            elif [ "${{ env.COMPILER }}" == "clang-cl" ]; then
+              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G Ninja -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_MODULE_PATH=build/generators/ -DCMAKE_PREFIX_PATH=build/generators/
```

```suggestion
              cmake -S . -B build -DCMAKE_BUILD_TYPE=$BUILD_TYPE -G "Visual Studio 17 2022" -T ClangCL
```
The ClangCL compiler is a part of MSVC, so it needs a VS project generator together with `ClangCL` MSVC toolset to be set to work correctly. See https://stackoverflow.com/questions/64189111/how-to-use-clang-cl-llvm-with-vs2019-cmake